ql/processes/piecewisetimedependenthestonprocess.cpp

```diff
@@ -0,0 +1,194 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+#include <ql/math/distributions/chisquaredistribution.hpp>
+#include <ql/math/distributions/normaldistribution.hpp>
+#include <ql/math/functional.hpp>
+#include <ql/math/integrals/gaussianquadratures.hpp>
+#include <ql/math/integrals/gausslobattointegral.hpp>
+#include <ql/math/integrals/segmentintegral.hpp>
+#include <ql/math/modifiedbessel.hpp>
+#include <ql/math/solvers1d/brent.hpp>
+#include <ql/processes/eulerdiscretization.hpp>
+#include <ql/processes/piecewisetimedependenthestonprocess.hpp>
+#include <ql/quotes/simplequote.hpp>
+#include <boost/math/distributions/non_central_chi_squared.hpp>
+#include <complex>
+#include <utility>
+
+namespace QuantLib {
+
+    PiecewiseTimeDependentHestonProcess::PiecewiseTimeDependentHestonProcess(
```

Comment:
Lots of duplication between here and the constant-parameter process, but I'm not sure how to remove it.
I'd probably move this to somewhere in the `ql/experimental` folder so we can change it freely later.